projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -176,7 +179,28 @@ public IAsyncBasicConsumer? DefaultConsumer
         [MemberNotNullWhen(false, nameof(CloseReason))]
         public bool IsOpen => CloseReason is null;
 
-        public ulong NextPublishSeqNo { get; private set; }
+        public ulong NextPublishSeqNo
```

Comment:
Is it a good idea to acquire the semaphore in a property getter and potentially block for a long time? Seems to be a very surprising side effect for a property.

It sounds like this should be a method returning a `ValueTask<ulong>`

Reply:
See https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1675